### PR TITLE
Adding comments and cleaning up whitespace

### PR DESCRIPTION
**Pull request by edhartnett**
I am adding comments and cleaning up whitespace issues as I study this code.

More to come, but let's get these changes merged...

No code changes in this PR, except whitespace stuff.

Still plenty of "???" in the code - feel free to use the code review to help me fill in missing comments.

Also took out all casts of the return value of bget(). Since bget() returns void *, there is no need to cast it. It will auto-cast to whatever you assign it to.

### CODE REVIEW

File: src/clib/pio_spmd.c [hunk 1]
```diff
@@ -328,134 +370,162 @@ int pio_swapm(void *sndbuf,   int sndlths[], int sdispls[],  MPI_Datatype stypes
...
+			       __FILE__,__LINE__);
+	    else
                 CheckMPIReturn(MPI_Send(ptr, sndlths[p], stypes[p], p, tag, comm), __FILE__,__LINE__);
-            }
-
         }
-        if(istep > maxreqh){
+
+	/* ??? */
+        if (istep > maxreqh)
```
Comment (jedwards4b) [inline src/clib/pio_spmd.c]: we did comms in sets of size max_reqs, if istep > maxreqh then there is a remainder that must be handled.

File: src/clib/pio_spmd.c [hunk 2]
```diff
@@ -198,99 +215,125 @@ int pio_fc_gatherv( void *sendbuf, const int sendcnt, const MPI_Datatype sendtyp
...
-    int maxrecv=0;
+    int maxsend = 0;
+    int maxrecv = 0;
 
     CheckMPIReturn(MPI_Comm_size(comm, &nprocs),__FILE__,__LINE__);
     CheckMPIReturn(MPI_Comm_rank(comm, &mytask),__FILE__,__LINE__);
 
-
-    if(max_requests == 0) {
+    if (max_requests == 0)
```
Comment (jedwards4b) [inline src/clib/pio_spmd.c]: if max_requests == 0 no throttling is requested and the default mpi_alltoallw function is used.

File: src/clib/pio_darray.c [hunk 3]
```diff
@@ -367,7 +368,7 @@ int PIOc_write_darray(const int ncid, const int vid, const int ioid,
     if ((recordvar && wmb->ioid != ioid) || (!recordvar && wmb->ioid != -(ioid)))
     {
 	/* Allocate a buffer. */
-        if (!(wmb->next = (wmulti_buffer *)bget((bufsize)sizeof(wmulti_buffer))))
```
Comment (edhartnett) [inline src/clib/pio_darray.c]: No need to typecast calls to bget. It returns void *.

File: src/clib/pio_rearrange.c [hunk 4]
```diff
@@ -820,88 +808,89 @@ int rearrange_comp2io(const iosystem_desc_t ios, io_desc_t *iodesc, void *sbuf,
...
         recvcounts[i] = 0;
         sdispls[i] = 0;
         rdispls[i] = 0;
-        sendtypes[ i ] = MPI_DATATYPE_NULL;
-        recvtypes[ i ] = MPI_DATATYPE_NULL;
+        sendtypes[i] = MPI_DATATYPE_NULL;
+        recvtypes[i] = MPI_DATATYPE_NULL;
     }
+
+    /* In IO tasks ??? */
```
Comment (edhartnett) [inline src/clib/pio_rearrange.c]: Clearly what? 

THe question is, in iotasks, the following code does what?

File: src/clib/pio_rearrange.c [hunk 5]
```diff
@@ -984,6 +968,7 @@ void determine_fill(iosystem_desc_t ios, io_desc_t *iodesc, const int gsize[],
 
     MPI_Allreduce(MPI_IN_PLACE, &totalllen, 1, PIO_OFFSET, MPI_SUM, ios.union_comm);
 
+    /* ??? */
```
Comment (edhartnett) [inline src/clib/pio_rearrange.c]: Did you see my other comment about how you should not be doing this with netCDF-4 files?

**Comment by edhartnett** [comment]
OK, thanks for the feedback. I have incorporated your comments into the code. I will leave these changes up a bit longer in case anyone else wants to look, then I will merge to master.

**Pull request merged**